src/drag-drop-context.tsx

```diff
@@ -1,30 +1,129 @@
+import {
+  layoutsAreEqual,
+  mostIntersectingLayout,
+  elementLayout,
+  transformLayout,
+  noopTransform,
+  Layout,
+  Transform,
+} from "./layout";
 import {
   batch,
+  Component,
   createContext,
   createEffect,
   mergeProps,
+  PropsWithChildren,
   untrack,
   useContext,
 } from "solid-js";
-import { createStore } from "solid-js/store";
-
-import {
-  layoutsAreEqual,
-  mostIntersectingLayout,
-  elementLayout,
-  transformLayout,
-  noopTransform,
-} from "./layout";
-
-export const Context = createContext();
-
-export const DragDropContext = (props) => {
-  props = mergeProps(
+import { createStore, Store } from "solid-js/store";
+
+type SensorActivator<K extends keyof HTMLElementEventMap> = (params: {
+  event: HTMLElementEventMap[K];
+  draggableId: string;
+}) => void;
+type Draggable = {
+  id: string;
```

Comment:
Question: Is there any reason to limit this to `string`? In some of my previous examples I happily used a `number` etc.

Reply:
My brain sees `id` it types `string` 😄 Changed that.